### ISSUE TITLE
Random Forest classifier does not work correctly with scikit >= 1.4.1

### ISSUE THREAD

**Issue by vkozmik** [iamDecode/sklearn-pmml-model#61]
Connected with the change in scikit, I think that the values of "votes" should be divided by "node_count"
API Change The tree_.value attribute in [tree.DecisionTreeClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.tree.DecisionTreeClassifier.html#sklearn.tree.DecisionTreeClassifier), [tree.DecisionTreeRegressor](https://scikit-learn.org/stable/modules/generated/sklearn.tree.DecisionTreeRegressor.html#sklearn.tree.DecisionTreeRegressor), [tree.ExtraTreeClassifier](https://scikit-learn.org/stable/modules/generated/sklearn.tree.ExtraTreeClassifier.html#sklearn.tree.ExtraTreeClassifier) and [tree.ExtraTreeRegressor](https://scikit-learn.org/stable/modules/generated/sklearn.tree.ExtraTreeRegressor.html#sklearn.tree.ExtraTreeRegressor) changed from an weighted absolute count of number of samples to a weighted fraction of the total number of samples. [#27639](https://github.com/scikit-learn/scikit-learn/pull/27639) by [Samuel Ronsin](https://github.com/samronsin).

Suggest to change line 276 of tree.py:

Originally:
votes = [[[float(votes_for(c).get('recordCount')) if votes_for(c) is not None else 0.0 for c in classes]]]

Suggested:
votes = [[[float(votes_for(c).get('recordCount')) / node_count if votes_for(c) is not None else 0.0 for c in classes]]]

**Comment by iamDecode** [comment]
Great, thanks for figuring this out! I would have to check if changing it to a weighted fraction is still compatible with older versions of scikit-learn.

**Comment by vkozmik** [comment]
You're welcome. Alternatively, we can wrap the new code in some condition for scikit version

major, minor, patch = map(int, sklearn.__version__.split("."))
if major <= 1 and (minor <= 3 or (major == 4 and patch ==0)):
    **old code**
else
   **new code**
   
not nice, but probably solves the issue

**Comment by iamDecode** [comment]
My bad, it took me a while to get back to this library. I've looked into the issue, and while I agree your suggested change makes sense, it does not quite work for this library. sklearn-pmml-model partially replaces the internal cython datastructure of trees to support categorical variables. Those internals are taken from a fork of sklearn that was unfortunately never merged. This means the Tree we use is not updated along with newer sklearn versions, and we still require non-normalized votes in trees.

I did implement your suggested change locally, but unfortunately this causes all tree based models to produce incorrect results according to our integration tests. Bottom line is that, even though the internal votes does not match exactly with the sklearn counterpart, the predictions are identical and we don't need any changes for this.

Im closing this issue since I don't think further changes are needed, but feel free to continue the discussion here.
